File: app/lib/pii.ts
const EMAIL_RE = /[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}/g;
const PHONE_RE =
  /(?:\+\d{1,3}[\s-]?)?(?:\(?\d{2,4}\)?[\s-]?)?\d{3}[\s-]?\d{2}[\s-]?\d{2}/g;

export function stripPii(text: string, names?: string[]): string {
  let result = text.replace(EMAIL_RE, "[EMAIL]").replace(PHONE_RE, "[PHONE]");
  for (const name of names ?? []) {
    if (name && name.length >= 2) {
      result = result.replaceAll(name, "[NAME]");
    }
  }
  return result;
}

File: worker/src/flatfox_worker/pii.py
import re

EMAIL_RE = re.compile(r"[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}")
PHONE_RE = re.compile(
    r"(?:\+\d{1,3}[\s\-]?)?"
    r"(?:\(?\d{2,4}\)?[\s\-]?)?"
    r"\d{3}[\s\-]?\d{2}[\s\-]?\d{2}"
)


def strip_pii(text: str, names: list[str] | None = None) -> str:
    result = EMAIL_RE.sub("[EMAIL]", text)
    result = PHONE_RE.sub("[PHONE]", result)
    for name in names or []:
        if name and len(name) >= 2:
            result = result.replace(name, "[NAME]")
    return result
